src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs

```diff
@@ -222,23 +222,20 @@ private static bool IsHostOrRemoteWorkspace(Project project)
 
         // Defer to the ProducerConsumer.  We're search the unreferenced projects in parallel. As we get results, we'll
         // add them to the 'allSymbolReferences' queue.  If we get enough results, we'll cancel all the other work.
-        await ProducerConsumer<ImmutableArray<SymbolReference>>.RunAsync(
-            ProducerConsumerOptions.SingleReaderOptions,
-            produceItems: static (onItemsFound, args) => RoslynParallel.ForEachAsync(
-                args.viableUnreferencedProjects,
-                args.linkedTokenSource.Token,
-                async (project, cancellationToken) =>
-                {
-                    // Search in this unreferenced project.  But don't search in any of its' direct references.  i.e. we
-                    // don't want to search in its metadata references or in the projects it references itself. We'll be
-                    // searching those entities individually.
-                    var references = await args.finder.FindInSourceSymbolsInProjectAsync(
-                        args.projectToAssembly, project, args.exact, cancellationToken).ConfigureAwait(false);
-                    onItemsFound(references);
-                }),
+        await ProducerConsumer<ImmutableArray<SymbolReference>>.RunParallelAsync(
+            viableUnreferencedProjects,
+            produceItems: static async (project, onItemsFound, args) =>
```

Comment:
this form is much easier to understand (for me at least).  instead of hte nested lambdas, you just pass hte single lambda.  It will be called in parallel, passing a single item from "viableUnreferencedProjects", and teh callback to call when items are produced based on htat item.